.github/workflows/ci-tests.yml

```diff
@@ -32,40 +32,49 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-22.04
-          - macos-12
-          - windows-2022
+        os: ${{ fromJson(github.repository_owner == 'subspace' && github.repository == 'subspace/subspace-cli' && '[["self-hosted", "ubuntu-22.04-x86-64", "Linux"], ["self-hosted", "macos-12-arm64", "macOS"], ["self-hosted", "windows-server-2022-x86-64", "Windows"]]' || '["ubuntu-22.04", "macos-12", "windows-2022"]') }}
         run-all:
           - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
         exclude: # exclude macos-12 and windows-2022 when the condition is false
           - run-all: false
             os: macos-12
           - run-all: false
             os: windows-2022
+          - run-all: false
+            os: macOS
+          - run-all: false
+            os: Windows
 
     runs-on: ${{ matrix.os }}
     steps:
       - name: git checkout
         uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # @v3.1.0
 
-      # On macOS, we need a proper Clang version, not Apple's custom version without wasm32 support
+      # On macOS, we need a proper Clang version, not Apple's custom version without wasm32 support, we also install clang on Windows
       - name: Install LLVM and Clang
         uses: KyleMayes/install-llvm-action@8852e4d5c58653ed05135c0a5d949d9c2febcb00 # v1.6.1
         with:
           version: "15.0"
-        if: runner.os == 'macOS'
+        if: runner.os == 'macOS' || runner.os == 'Windows'
 
       - name: Install Protoc
-        uses: arduino/setup-protoc@v1
+        uses: arduino/setup-protoc@v2
         with:
           repo-token: ${{ secrets.GITHUB_TOKEN }}
+        if: runner.os == 'Linux' || runner.os == 'macOS'
 
-      # Workaround to resolve link error with C:\msys64\mingw64\bin\libclang.dll
-      - name: Remove msys64
-        run: Remove-Item -LiteralPath "C:\msys64\" -Force -Recurse
+      - name: Install Protoc Windows
+        run: |
+          Invoke-WebRequest -Uri https://github.com/protocolbuffers/protobuf/releases/download/v23.3/protoc-23.3-win64.zip -Outfile protoc-23.3-win64.zip
+          Expand-Archive -Force .\protoc-23.3-win64.zip -DestinationPath C:\protoc-23.3 
         if: runner.os == 'Windows'
```

Comment:
Why two ways to install `protoc`? It should be the same way it was before. If it doesn't work then runner needs to be fixed.

Reply:
Also you removed "Remove msys64" step, which means you broke compatibility with GitHub's runners, while I mentioned many times it must continue to be supported like it was before.

Reply:
The `msys64` package part was causing the windows jobs to fail since the directory didn't exist. The package is used to install clang, I will add it back and include an option to only execute the step when runner is not self-hosted.

Reply:
For `protoc` the actions was failing on windows and I reported the bug upstream, so I used another method for Windows. However, I've installed `protoc`  on the windows machine and added it to the path, will test it and remove the step.

Reply:
If `msys64` doesn't exist on self-hosted workers there are two options:
1) To install it such that it exists just like of GitHub's runners
2) Allow the step to fail with [`continue-on-error: true`](https://github.com/subspace/subspace/commit/9bc0e22af33f3eace37ecbedd958a55354c3452a#diff-73e17259d77e5fbef83b2bdbbe4dc40a912f807472287f7f45b77e0cbf78792dR88-R89)

Removing it and breaking compatibility with GitHub runners is not a desired solution.

Reply:
If `protoc` is not installed on GitHub's runners, it should not be installed on our self-hosted runners either. If there is an issue upstream, make sure to leave a TODO in the code linking to upstream bug report/PR so we can track the progress and/or influence the timeline for fixing it.

Right now there is no comment, no link, reader would have no idea why it is necessary.

Reply:
Fixed msys64 to work with github runners only, it can't work with self-hosted runners since we don't use the package for clang.

For protoc it fails with
```
Run arduino/setup-protoc@v2
Getting protoc version: v23.4
Downloading archive: https://github.com/protocolbuffers/protobuf/releases/download/v23.4/protoc-23.4-win64.zip
C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoLogo -Sta -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command "$ErrorActionPreference = 'Stop' ; try { Add-Type -AssemblyName System.IO.Compression.FileSystem } catch { } ; if ((Get-Command -Name Expand-Archive -Module Microsoft.PowerShell.Archive -ErrorAction Ignore)) { Expand-Archive -LiteralPath 'C:\actions-runner\_work\_temp\4a204b5d-f617-437a-bb54-c2a7572224a7' -DestinationPath 'C:\actions-runner\_work\_temp\de640d4f-9cdb-43db-b478-499344c9c0c6' -Force } else {[System.IO.Compression.ZipFile]::ExtractToDirectory('C:\actions-runner\_work\_temp\4a204b5d-f617-437a-bb54-c2a7572224a7', 'C:\actions-runner\_work\_temp\de640d4f-9cdb-43db-b478-499344c9c0c6', $true) }"
Expand-Archive :  is not a supported archive file format. .zip is the only supported archive file format.
```

this is because it doesn't expand the path to include the `.zip` extension. The bug needs to be patched as discussed in https://github.com/arduino/setup-protoc/issues/86

Reply:
i have installed protoc on the windows runner directly, so that extra step has been removed.

Reply:
> since we don't use the package for clang and use visualstudio 2022 c++ tools

As commeneted above we need community version of Visual Studio 2022 *AND* poroper Clang/LLVM from upstream release. I don't know what version comes with Visual Studio, might be even MSVC with Clang CLI, not actual Clang.

Reply:
> i have installed protoc on the windows runner directly, so that extra step has been removed.


Then you should remove it, it is not present on GitHub's runners and we should not be installed on our runners as well. There should be no extra software on self-hosted runners that is not on GitHub's runners that our workflows rely on.